js_modules/dagster-ui/packages/ui-core/src/asset-selection/input/util.ts

```diff
@@ -66,3 +63,20 @@ export const getAttributesMap = (assets: AssetGraphQueryItem[]) => {
     code_location: codeLocations,
   };
 };
+
+const memoizedTag = weakMapMemoize((key: string, value: string) => ({
+  key,
+  value,
+}));
```

Comment:
It seems like getAttributesMap should be memoized as a whole (and it is in the hook). I wonder if this is really necessary? I guess this makes the tag objects identities remain the same across changes in the `[assets]` deps in the hook, but that seems like it might be a rare case.